pkg/project/git.go

```diff
@@ -137,44 +138,24 @@ func (g Git) Detect() (Result, bool, error) {
 		return Result{
 			Project: project,
 			Branch:  branch,
-			Folder:  gitdir,
+			Folder:  filepath.Join(gitdir, ".."),
 		}, true, nil
 	}
 
 	return Result{}, false, nil
 }
 
-func findGitConfigFile(fp, dir, match string) (string, bool) {
-	i := 0
-	for i < maxRecursiveIteration {
-		if fileExists(filepath.Join(fp, dir, match)) {
-			return filepath.Join(fp, dir), true
-		}
-
-		fp = filepath.Clean(filepath.Join(fp, ".."))
-		if fp == "." || fp == "/" || driveLetterRegex.MatchString(fp) {
-			return "", false
-		}
-
-		i++
-	}
-
-	log.Warnf("didn't find git config file after %d iterations", maxRecursiveIteration)
-
-	return "", false
-}
-
 func findSubmodule(fp string, patterns []regex.Regex) (string, bool, error) {
 	if !shouldTakeSubmodule(fp, patterns) {
 		return "", false, nil
 	}
 
-	gitConfigFile, ok := findGitConfigFile(fp, "", ".git")
+	gitConfigFile, ok := findFileOrDirectory(fp, "", ".git")
```

Comment:
This is actually the .git directory, not the git config file.
```suggestion
	gitDir, ok := findFileOrDirectory(fp, "", ".git")
```